### PR TITLE
Style fixes for searchcorespi::index::IndexMaintainer.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

**Review by baldersheim — DISMISSED** [review]

File: searchcore/src/vespa/searchcorespi/index/indexmaintainer.cpp [hunk 1]
```diff
@@ -970,7 +969,7 @@ IndexMaintainer::initFlush(SerialNum serialNum, searchcorespi::FlushStats * stat
         set_current_serial_num(std::max(current_serial_num(), serialNum));
     }
 
-    IMemoryIndex::SP new_index(_operations.createMemoryIndex(getSchema(), *_current_index, current_serial_num()));
+    std::shared_ptr<IMemoryIndex> new_index(_operations.createMemoryIndex(getSchema(), *_current_index, current_serial_num()));
```
Comment (geirst) [inline searchcore/src/vespa/searchcorespi/index/indexmaintainer.cpp:972]: Consider using auto instead.

File: searchcore/src/vespa/searchcorespi/index/indexmaintainer.cpp [hunk 2]
```diff
@@ -1115,12 +1114,12 @@ IndexMaintainer::runFusion(const FusionSpec &fusion_spec, std::shared_ptr<search
     }
 
     const string new_fusion_dir = getFusionDir(new_fusion_id);
-    Schema::SP prunedSchema = getActiveFusionPrunedSchema();
+    std::shared_ptr<Schema> prunedSchema = getActiveFusionPrunedSchema();
```
Comment (geirst) [inline searchcore/src/vespa/searchcorespi/index/indexmaintainer.cpp:1117]: Consider using auto instead.

File: searchcore/src/vespa/searchcorespi/index/indexmaintainer.cpp [hunk 3]
```diff
@@ -1115,12 +1114,12 @@ IndexMaintainer::runFusion(const FusionSpec &fusion_spec, std::shared_ptr<search
...
     if (prunedSchema) {
         updateDiskIndexSchema(new_fusion_dir, *prunedSchema, noSerialNumHigh);
     }
     ChangeGens changeGens = getChangeGens();
-    IDiskIndex::SP new_index(loadDiskIndex(new_fusion_dir));
+    std::shared_ptr<IDiskIndex> new_index(loadDiskIndex(new_fusion_dir));
```
Comment (geirst) [inline searchcore/src/vespa/searchcorespi/index/indexmaintainer.cpp:1122]: Consider using auto instead.

File: searchcore/src/vespa/searchcorespi/index/indexmaintainer.cpp [hunk 4]
```diff
@@ -1315,7 +1314,7 @@ IndexMaintainer::setSchema(const Schema & schema, SerialNum serialNum)
 {
     assert(_ctx.getThreadingService().master().isCurrentThread());
     pruneRemovedFields(schema, serialNum);
-    IMemoryIndex::SP new_index(_operations.createMemoryIndex(schema, *_current_index, current_serial_num()));
+    std::shared_ptr<IMemoryIndex> new_index(_operations.createMemoryIndex(schema, *_current_index, current_serial_num()));
```
Comment (geirst) [inline searchcore/src/vespa/searchcorespi/index/indexmaintainer.cpp:1317]: Consider using auto instead.

File: searchcore/src/vespa/searchcorespi/index/indexmaintainer.cpp [hunk 5]
```diff
@@ -1331,8 +1330,8 @@ void
 IndexMaintainer::pruneRemovedFields(const Schema &schema, SerialNum serialNum)
 {
     assert(_ctx.getThreadingService().master().isCurrentThread());
-    ISearchableIndexCollection::SP new_source_list;
-    IIndexCollection::SP coll = getSourceCollection();
+    std::shared_ptr<ISearchableIndexCollection> new_source_list;
+    std::shared_ptr<IIndexCollection> coll = getSourceCollection();
```
Comment (geirst) [inline searchcore/src/vespa/searchcorespi/index/indexmaintainer.cpp:1334]: Consider using auto instead.

**Review by geirst — DISMISSED** [review]
👍

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**